rules-reviewed/jboss-eap/websphere/WebSphere.windup.xml

```diff
@@ -0,0 +1,1219 @@
+                    <location>INHERITANCE</location>
+                </javaclass>
+            </when>
+            <perform>
+                <classification classification="WebSphere Startup Service" effort="4">
+                    <link href="http://docs.oracle.com/javaee/6/api/javax/ejb/Singleton.html" description="EJB 3.1 Singleton Bean"/>
+                    <link href="http://docs.oracle.com/javaee/6/api/javax/ejb/Startup.html" description="EJB 3.1 Startup Bean"/>
+                </classification>
+                <hint>
+                    WebSphere Startup Service can be replaced with EJB 3.1 Startup Bean, using `@Startup @Singleton` class and `@PostConstruct` method.
+                </hint>
+            </perform>
+        </rule>
+        <rule id="WebSphere>eap6:-Altni">
+            <when>
+                <javaclass references="com.ibm.websphere.startupservice.AppStartUpHome" as="default"/>
+            </when>
+            <perform>
+                <hint message="Replace with EJB 3.1 @Singleton / @Startup annotations." effort="0">
+                    <link href="http://docs.oracle.com/javaee/6/api/javax/ejb/Singleton.html" description="EJB 3.1 Singleton Bean"/>
+                    <link href="http://docs.oracle.com/javaee/6/api/javax/ejb/Startup.html" description="EJB 3.1 Startup Bean"/>
+                </hint>
+            </perform>
+        </rule>
+
+        <!-- IBM JMS MQe -->
+        <rule id="WebSphere>eap6:-Epnmk">
+            <when>
+                <javaclass references="com.ibm.mqe.jms.MQeConnection" as="default"/>
+            </when>
+            <perform>
+                <classification description="Legacy IBM MQe API. A more general, and recommended, approach is to use the Connection class.">
+                    <link description="Javadoc" href="http://www-01.ibm.com/support/knowledgecenter/SSVHEW_6.2.0/com.ibm.mqe.javadoc/java/com/ibm/mqe/jms/MQeQueueConnection.html"></link>
+                </classification>
+                <hint message="Replace with Java EE 6 JMS: `javax.jms.Connection`." effort="1">
+                    <link description="Java EE 6 JMS Tutorial" href="https://docs.oracle.com/javaee/6/tutorial/doc/bnceh.html"/>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="WebSphere>eap6:MQeQC-10vvyf">
+            <when>
+                <javaclass references="com.ibm.mqe.jms.MQeQueueConnection" as="default"/>
+            </when>
+            <perform>
+                <classification description="Legacy IBM MQe API. A more general, and recommended, approach is to use the QueueConnection class.">
+                    <link description="Javadoc" href="http://www-01.ibm.com/support/knowledgecenter/SSVHEW_6.2.0/com.ibm.mqe.javadoc/java/com/ibm/mqe/jms/MQeQueueConnection.html"></link>
+                </classification>
+                <hint message="Replace with Java EE 6 JMS: `javax.jms.QueueConnection`." effort="1">
```

Comment:
I think that is a good idea, although it might also be a good idea to split all of the related rules into their own ruleset (something like websphere-jms.windup.xml).